### PR TITLE
Warnint 64to32 6186 v8

### PR DESCRIPTION
**Pull request by catenacyber**
[Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6186

Describe changes:
- fix `-Wshorten-64-to-32` warnings for some files

```
LIBHTP_BRANCH=pr/399
```
https://github.com/OISF/libhtp/pull/399

#9503 with more fixes

Still more warnings to fix and to add in CI
- detect* files
- u* files

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on ASAN_TLPR1_suri.
  

Pipeline 16038

File: src/app-layer-htp-range.c [hunk 1]
```diff
@@ -543,14 +543,24 @@ File *HttpRangeClose(const StreamingBufferConfig *sbcfg, HttpRangeContainerBlock
             // a new range just begins where we ended, append it
             if (range->gap > 0) {
                 // if the range had a gap, begin by it
-                if (FileAppendData(c->container->files, sbcfg, NULL, range->gap) != 0) {
+                uint32_t gap = (uint32_t)range->gap;
+                if (range->gap > UINT32_MAX) {
```
Comment (inashivb) [inline src/app-layer-htp-range.c:547]: If we are downcasting `range->gap` to `uint32_t` first, under what circumstances should it be greater than `UINT32_MAX`?

**Review by inashivb — COMMENTED** [review]
Hi, Philippe!
A question inline about a pattern I see throughout the PR.
I tried something trivial:
```c
#include <stdio.h>
#include <stdint.h>

int main()
{
    uint16_t myvar = 65535;
    printf("%d\n", myvar);
    uint8_t blah = (uint8_t)myvar;
    printf("%d\n", blah);
    if (blah > UINT8_MAX) {
        printf("YES\n");
    }
    return 0;
}
```
to understand how casting and limits work, with the max u16 value, I always get blah to be 255. Can something and if yes what would make blah cross 255?

File: src/app-layer-htp-range.c (hunk 1 above)
Comment (catenacyber) [reply src/app-layer-htp-range.c:547]: Indeed,
looks like `range->gap > UINT32_MAX` if we had multiple gaps which combine over 4G

Let's redefine `range->gap` to be u32 in the first place

Thanks

**Comment by catenacyber** [comment]
> Hi, Philippe! A question inline about a pattern I see throughout the PR. I tried something trivial:
> 
> to understand how casting and limits work, with the max u16 value, I always get blah to be 255. Can something and if yes what would make blah cross 255?

You should test myvar against 255 before casting it into `uint8_t blah`

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/9512

**Pull request closed without merging**